test/deprecated/parenful/not-procedure.good

```diff
@@ -0,0 +1 @@
+not-procedure.chpl:2: error: the '@deprecated' attribute with 'parenless=true' can only be applied to parenless procedures
```

Comment:
Your tests in test/deprecated/parenful really belong in `test/deprecated-keyword/`, please move them there. I suggest also adding a test of the default deprecation message, so that it is easier for somebody to realize that this functionality is available.

As far as `test/deprecated/`, please add a test of the deprecated uses of list.first() and .last() there. I suggest starting the code with something like `// deprecated in 1.32 by Daniel` .